### PR TITLE
images/metering-ansible-operator: Move all reporting-operator TLS-related tasks to its own task file and clean-up TLS-related tasks.

### PR DESCRIPTION
**Pull request by timflannagan1**
This would add another file `configure_reporting_operator_openssl.yml`, which is used to automatically generate client cert/key for the reporting-operator to use when communicating with Presto and Hive Server.

Also, in this PR, there was some inconsistency on how related tasks were handled between the reporting-operator, Presto, and Root CA. I'd rather make these changes in a PR separate from #825 .

### CODE REVIEW

File: images/metering-ansible-operator/roles/meteringconfig/tasks/configure_presto_tls.yml [hunk 1]
```diff
@@ -1,34 +1,55 @@
...
+  vars:
+    presto_tls_secret_exists: "{{ presto_secret_tls_buf.resources and presto_secret_tls_buf.resources | length > 0 }}"
+    presto_auth_secret_exists: "{{ presto_secret_auth_buf.resources and presto_secret_auth_buf.resources | length > 0 }}"
   when: meteringconfig_tls_enabled
 
 #
-# Generate server and client certificates using the Ansible OpenSSL modules when top-level spec.tls.enabled is set to true
+# Generate server and client certificates (as needed) using the Ansible OpenSSL modules when top-level spec.tls.enabled is set to true
 #
 - name: Configure TLS and client-side authentication for Presto and reporting-operator
```
Comment (chancez) [inline images/metering-ansible-operator/roles/meteringconfig/tasks/configure_presto_tls.yml]: Name needs updating to be just "presto"

Comment (timflannagan1) [reply images/metering-ansible-operator/roles/meteringconfig/tasks/configure_presto_tls.yml]: Updated.

File: images/metering-ansible-operator/roles/meteringconfig/tasks/validate_reporting_operator_tls.yml [hunk 2]
```diff
@@ -23,4 +23,3 @@
         - meteringconfig_spec['reporting-operator'].spec.authProxy.authenticatedEmails.data != ""
       msg: "spec.reporting-operator.spec.authProxy.authenticatedEmails: secretName and/or data key(s) cannot be empty when enabled and createSecret is set to true"
     when: meteringconfig_spec['reporting-operator'].spec.authProxy.authenticatedEmails.enabled and meteringconfig_spec['reporting-operator'].spec.authProxy.authenticatedEmails.createSecret
-  when: not meteringconfig_tls_enabled
```
Comment (chancez) [inline images/metering-ansible-operator/roles/meteringconfig/tasks/validate_reporting_operator_tls.yml]: Won't this break when TLS is disabled then?

File: images/metering-ansible-operator/roles/meteringconfig/tasks/configure_reporting_operator_tls.yml [hunk 3]
```diff
@@ -3,63 +3,80 @@
 #
 # Presto and reporting-operator TLS
```
Comment (chancez) [inline images/metering-ansible-operator/roles/meteringconfig/tasks/configure_reporting_operator_tls.yml]: Comment could be updated

File: images/metering-ansible-operator/roles/meteringconfig/tasks/validate_reporting_operator_tls.yml (hunk 2 above)
Comment (timflannagan1) [reply images/metering-ansible-operator/roles/meteringconfig/tasks/validate_reporting_operator_tls.yml]: Nope - this at was redundant:
```
- name: Validate the user-provided authProxy configuration
  include_tasks: validate_reporting_operator_tls.yml
  when: not meteringconfig_tls_enabled
```

Comment (timflannagan1) [reply images/metering-ansible-operator/roles/meteringconfig/tasks/validate_reporting_operator_tls.yml]: Nope - that was redundant:

- name: Validate the user-provided authProxy configuration
  include_tasks: validate_reporting_operator_tls.yml
  when: not meteringconfig_tls_enabled

File: images/metering-ansible-operator/roles/meteringconfig/tasks/configure_reporting_operator_tls.yml (hunk 3 above)
Comment (timflannagan1) [reply images/metering-ansible-operator/roles/meteringconfig/tasks/configure_reporting_operator_tls.yml]: Updated

File: images/metering-ansible-operator/roles/meteringconfig/tasks/validate_reporting_operator_tls.yml (hunk 2 above)
Comment (chancez) [reply images/metering-ansible-operator/roles/meteringconfig/tasks/validate_reporting_operator_tls.yml]: Ah

**Pull request merged**